### PR TITLE
AeroJEPA: stable checkpoint dir + automatic resume (fix Hydra timestamp resume)

### PR DESCRIPTION
**Pull request by mnabian**
# PhysicsNeMo Pull Request

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: examples/cfd/external_aerodynamics/aerojepa/train.py [hunk 1]
```diff
@@ -746,6 +747,20 @@ def _save_checkpoint(
...
+
+    The ``epoch_XXXX.pt`` names are zero-padded, so a lexicographic sort
+    matches numeric order. Used to resume automatically from a stable
+    checkpoint directory.
+    """
+    ckpt_dir = Path(ckpt_dir)
+    if not ckpt_dir.is_dir():
+        return None
+    ckpts = sorted(ckpt_dir.glob("epoch_*.pt"))
+    return ckpts[-1] if ckpts else None
```
[the remark this answers is not in the document]
Comment (mnabian) [reply examples/cfd/external_aerodynamics/aerojepa/train.py]: Fixed.

File: examples/cfd/external_aerodynamics/aerojepa/train.py [hunk 2]
```diff
@@ -754,51 +769,56 @@ def _load_initial_state(
...
     resume_cfg = train_cfg.get("resume", None)
+
+    # (1) explicit external path wins; else (2) latest in the stable ckpt_dir.
+    explicit = None
     if resume_cfg is not None and bool(resume_cfg.get("enabled", False)):
-        ckpt_path = resume_cfg.get("checkpoint_path")
-        if not ckpt_path:
-            raise ValueError("training.resume.enabled=true requires checkpoint_path.")
+        explicit = resume_cfg.get("checkpoint_path")
+    resume_path = explicit or _latest_checkpoint(ckpt_dir)
```
[the remark this answers is not in the document]
Comment (mnabian) [reply examples/cfd/external_aerodynamics/aerojepa/train.py:816]: Fixed.

File: examples/cfd/external_aerodynamics/aerojepa/train.py [hunk 3]
```diff
@@ -746,6 +747,30 @@ def _save_checkpoint(
...
+    Selected by the integer epoch parsed from the filename (not a
+    lexicographic sort), so it stays correct past ``epoch_9999.pt`` and for
+    mixed zero-pad widths. Used to resume automatically from a stable
+    checkpoint directory.
+    """
+    ckpt_dir = Path(ckpt_dir)
+    if not ckpt_dir.is_dir():
+        return None
+    ckpts = list(ckpt_dir.glob("epoch_*.pt"))
+    return max(ckpts, key=_epoch_of) if ckpts else None
```
Comment (negin513) [inline examples/cfd/external_aerodynamics/aerojepa/train.py:773]: If every file in `ckpt_dir` has an unparseable stem (e.g. `epoch_abc.pt`), `_epoch_of` returns `-1` for all of them and `max()` returns an arbitrary file silently. Suggest filtering to only files with a valid (≥ 0) epoch before calling `max`:
```python
valid = [(p, _epoch_of(p)) for p in ckpts if _epoch_of(p) >= 0]
return max(valid, key=lambda x: x[1])[0] if valid else None
```

File: examples/cfd/external_aerodynamics/aerojepa/train.py [hunk 4]
```diff
@@ -754,51 +779,61 @@ def _load_initial_state(
...
-        ckpt_path = resume_cfg.get("checkpoint_path")
-        if not ckpt_path:
-            raise ValueError("training.resume.enabled=true requires checkpoint_path.")
+        explicit = resume_cfg.get("checkpoint_path")
+        if not explicit:
+            raise ValueError(
+                "training.resume.enabled=true requires checkpoint_path; omit "
+                "resume.enabled to auto-resume from the run's checkpoint dir."
+            )
+    resume_path = explicit or _latest_checkpoint(ckpt_dir)
```
Comment (negin513) [inline examples/cfd/external_aerodynamics/aerojepa/train.py:816]: Silent-override risk: if a user has configured `training.init_from_checkpoint` to warm-start from pretrained weights, but `ckpt_dir` already contains an `epoch_*.pt` from a prior aborted run, auto-resume wins here and `init_from_checkpoint` is never reached — they get a full resume instead of the pretrained-weights-only init they intended, with no warning. Suggest logging when this happens:
```python
if resume_path and train_cfg.get("init_from_checkpoint", {}).get("path"):
    log.warning(
        "Auto-resume from %s takes priority over init_from_checkpoint. "
        "Delete ckpt_dir to force a fresh pretrained init.", resume_path
    )
```

File: examples/cfd/external_aerodynamics/aerojepa/train.py [hunk 5]
```diff
@@ -754,51 +779,61 @@ def _load_initial_state(
...
-            and payload.get("lr_scheduler")
-        ):
+        if load_sched and lr_scheduler is not None and payload.get("lr_scheduler"):
             lr_scheduler.load_state_dict(payload["lr_scheduler"])
         if ema is not None and payload.get("ema_shadow"):
             ema.shadow = {k: v.to(device) for k, v in payload["ema_shadow"].items()}
         start_epoch = int(payload.get("epoch", 0))
         best_val = float(payload.get("best_val_loss", float("inf")))
         log.info(
             "Resumed from %s at epoch %d (best_val=%.4e)",
```
Comment (negin513) [inline examples/cfd/external_aerodynamics/aerojepa/train.py]: Both explicit and auto-resume hit this same log line, making it hard to distinguish in logs whether a job was explicitly pointed at a checkpoint or silently picked one up. Suggest tagging the source:
```python
mode = "explicit" if explicit else "auto"
log.info("[resume:%s] Resumed from %s at epoch %d (best_val=%.4e)", mode, resume_path, start_epoch, best_val)
```

File: examples/cfd/external_aerodynamics/aerojepa/conf/training/superwing.yaml [hunk 6]
```diff
@@ -21,7 +21,7 @@ batch_size: 1
 eval_batch_size: 1
 max_eval_batches: 4
 precision: bf16
-save_every_epochs: 50
+save_every_epochs: 10
```
Comment (negin513) [inline examples/cfd/external_aerodynamics/aerojepa/conf/training/superwing.yaml:24]: This quietly changed from 50 → 10, which is 5× more frequent checkpoint saves. For a long run that's a meaningful disk cost. Worth calling this out explicitly in the PR description and confirming it's intentional (vs. a debug leftover).

**Review by negin513 — APPROVED** [review]
Left a few inline notes (silent init_from_checkpoint override, log disambiguation, unparseable-stem edge case, save_every_epochs change) -  none are blockers, but worth a look before merge. Overall: good change, approving.

File: examples/cfd/external_aerodynamics/aerojepa/conf/training/superwing.yaml (hunk 6 above)
Comment (mnabian) [reply examples/cfd/external_aerodynamics/aerojepa/conf/training/superwing.yaml:24]: Yes, this was intentional.

File: examples/cfd/external_aerodynamics/aerojepa/train.py (hunk 4 above)
Comment (mnabian) [reply examples/cfd/external_aerodynamics/aerojepa/train.py:816]: Fixed.

File: examples/cfd/external_aerodynamics/aerojepa/train.py (hunk 5 above)
Comment (mnabian) [reply examples/cfd/external_aerodynamics/aerojepa/train.py]: Fixed.

File: examples/cfd/external_aerodynamics/aerojepa/train.py (hunk 3 above)
Comment (mnabian) [reply examples/cfd/external_aerodynamics/aerojepa/train.py:773]: Fixed
